lib/msf/core/exploit/remote/socket_server.rb

```diff
@@ -80,17 +82,24 @@ def start_service(opts = {})
   # Stops the service.
   #
   def stop_service
-    if (service)
+    if service
       begin
-        self.service.deref if self.service.kind_of?(Rex::Service)
-        if self.service.kind_of?(Rex::Socket) || self.service.kind_of?(Rex::Post::Meterpreter::Channel)
```

Comment:
I wonder if I'm missing something more nuanaced about the previous code? 🤔 

I'm guessing we just don't have any Rex::Socket / Rex::Post::Meterpreter::Channel instances that are also services